llvm/lib/Transforms/Scalar/BDCE.cpp

```diff
@@ -125,6 +125,28 @@ static bool bitTrackingDCE(Function &F, DemandedBits &DB) {
       }
     }
 
+    // Simplify `and` or `or` when their mask does not affect the demanded bits.
+    if (auto *BO = dyn_cast<BinaryOperator>(&I)) {
+      unsigned Opc = BO->getOpcode();
+      if (Opc == Instruction::And || Opc == Instruction::Or) {
```

Comment:
Could you turn it into a switch, please? I think there is room for other operators like `xor`, `add` to be here in the future.